The job **`aarch64-gnu-llvm-20-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/22265186012/job/64409962981) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/64409962981) [(plain)](https://github.com/rust-lang/rust/commit/abcdef06402d5d6a2d002acd841de9202902b59a/checks/64409962981/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
REPOSITORY                                   TAG       IMAGE ID       CREATED       SIZE
ghcr.io/dependabot/dependabot-updater-core   latest    afc745c7535d   3 weeks ago   783MB
=> Removing docker images...
Deleted Images:
untagged: ghcr.io/dependabot/dependabot-updater-core:latest
untagged: ghcr.io/dependabot/dependabot-updater-core@sha256:faae3d3a1dedd24cde388bb506bbacc0f7ed1eae99ebac129af66acd8540c84a
deleted: sha256:afc745c7535da1bb12f92c273b9a7e9c52c3f12c5873714b2542da259c6d9769
deleted: sha256:64e147d5e54d9be8b8aa322e511cda02296eda4b8b8d063c6a314833aca50e29
deleted: sha256:5cba409bb463f4e7fa1a19f695450170422582c1bc7c0e934d893b4e5f558bc6
deleted: sha256:cddc6ebd344b0111eaab170ead1dfda24acdfe865ed8a12599a34d338fa8e28b
deleted: sha256:2412c3f334d79134573cd45e657fb6cc0abd75bef3881458b0d498d936545c8d
---
test [ui] tests/ui/abi/abi-sysv64-register-usage.rs ... ignored, ignored when the architecture is aarch64
test [ui] tests/ui/abi/abi-typo-unstable.rs#feature_enabled ... ok
test [ui] tests/ui/abi/abi-typo-unstable.rs#feature_disabled ... ok
test [ui] tests/ui/abi/anon-extern-mod.rs ... ok
test [ui] tests/ui/abi/avr-sram.rs#disable_sram ... ok
test [ui] tests/ui/abi/avr-sram.rs#has_sram ... ok
test [ui] tests/ui/abi/arm-unadjusted-intrinsic.rs#aarch64 ... ok
test [ui] tests/ui/abi/avr-sram.rs#no_sram ... ok
test [ui] tests/ui/abi/arm-unadjusted-intrinsic.rs#arm ... ok
test [ui] tests/ui/abi/bad-custom.rs ... ok
test [ui] tests/ui/abi/c-stack-as-value.rs ... ok
test [ui] tests/ui/abi/c-zst.rs#aarch64-darwin ... ok
test [ui] tests/ui/abi/c-zst.rs#powerpc-linux ... ok
---
test [ui] tests/ui/asm/aarch64/type-check-2.rs ... ok
test [ui] tests/ui/asm/aarch64/ttbr0_el2.rs ... ok
test [ui] tests/ui/asm/aarch64/type-check-3.rs ... ok
test [ui] tests/ui/asm/aarch64/type-f16.rs ... ok
test [ui] tests/ui/asm/aarch64v8r.rs#r82 ... ok
test [ui] tests/ui/asm/aarch64v8r.rs#hf ... ok
test [ui] tests/ui/asm/aarch64/sym.rs ... ok
test [ui] tests/ui/asm/aarch64v8r.rs#sf ... ok
test [ui] tests/ui/asm/arm-low-dreg.rs ... ok
test [ui] tests/ui/asm/binary_asm_labels.rs ... ignored, only executed when the architecture is x86_64
test [ui] tests/ui/asm/asm-with-nested-closure.rs ... ok
test [ui] tests/ui/asm/cfg-parse-error.rs ... ok
test [ui] tests/ui/asm/cfg.rs#reva ... ignored, only executed when the architecture is x86_64
---
test [ui] tests/ui/const-generics/occurs-check/unused-substs-1.rs ... ok
test [ui] tests/ui/const-generics/ogca/basic.rs ... ok
test [ui] tests/ui/const-generics/occurs-check/unused-substs-4.rs ... ok
test [ui] tests/ui/const-generics/occurs-check/unused-substs-5.rs ... ok
test [ui] tests/ui/const-generics/ogca/basic-fail.rs ... ok
test [ui] tests/ui/const-generics/ogca/coherence-ambiguous.rs ... ok
test [ui] tests/ui/const-generics/ogca/rhs-but-not-root.rs ... ok
test [ui] tests/ui/const-generics/opaque_types.rs ... ok
test [ui] tests/ui/const-generics/opaque_types2.rs ... ok
test [ui] tests/ui/const-generics/outer-lifetime-in-const-generic-default.rs ... ok
test [ui] tests/ui/const-generics/overlapping_impls.rs ... ok
test [ui] tests/ui/const-generics/params-in-ct-in-ty-param-lazy-norm.rs#full ... ok
---
test [ui] tests/ui/extern/issue-64655-extern-rust-must-allow-unwind.rs#fat2 ... ok
test [ui] tests/ui/extern/issue-64655-extern-rust-must-allow-unwind.rs#fat3 ... ok
test [ui] tests/ui/extern/issue-80074.rs ... ok
test [ui] tests/ui/extern/issue-95829.rs ... ok
test [ui] tests/ui/extern/lgamma-linkage.rs ... ok
test [ui] tests/ui/extern/issue-64655-extern-rust-must-allow-unwind.rs#thin1 ... ok
test [ui] tests/ui/extern/no-mangle-associated-fn.rs ... ok
test [ui] tests/ui/extern/not-in-block.rs ... ok
test [ui] tests/ui/extern/unsized-extern-derefmove.rs ... ok
test [ui] tests/ui/feature-gates/allow-features-empty.rs ... ok
---
test [ui] tests/ui/feature-gates/feature-gate-macro-derive.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-macro-metavar-expr-concat.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-marker_trait_attr.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-may-dangle.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-mgca-type-const-syntax.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-min-generic-const-args.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-movrs_target_feature.rs ... ignored, only executed when the architecture is x86_64
test [ui] tests/ui/feature-gates/feature-gate-more-maybe-bounds.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-min_const_fn.rs ... ok
test [ui] tests/ui/feature-gates/feature-gate-naked_functions_rustic_abi.rs ... ignored, only executed when the architecture is x86_64
---
test [ui] tests/ui/imports/ambiguous-9.rs ... ok
test [ui] tests/ui/imports/ambiguous-import-visibility-module.rs ... ok
test [ui] tests/ui/imports/ambiguous-8.rs ... ok
test [ui] tests/ui/imports/ambiguous-glob-vs-expanded-extern.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-glob-vs-multiouter.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-globvsglob.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-no-implicit-prelude.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-non-prelude-core-glob.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-non-prelude-std-glob.rs ... ok
test [ui] tests/ui/imports/ambiguous-import-visibility-macro.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-pick-core.rs ... ok
---
test [ui] tests/ui/layout/post-mono-layout-cycle.rs ... ok
test [ui] tests/ui/layout/randomize.rs#normal ... ok
test [ui] tests/ui/layout/randomize.rs#randomize-layout ... ok
test [ui] tests/ui/layout/null-pointer-optimization.rs ... ok
test [ui] tests/ui/layout/rigid-alias-due-to-broken-impl.rs ... ok
test [ui] tests/ui/layout/size-of-val-raw-too-big.rs ... ignored, only executed when the pointer width is 32bit (Layout computation rejects this layout for different reasons on 64-bit.)
test [ui] tests/ui/layout/reprc-power-alignment.rs ... ok
test [ui] tests/ui/layout/struct.rs ... ok
test [ui] tests/ui/layout/rust-call-abi-not-a-tuple-ice-81974.rs ... ok
test [ui] tests/ui/layout/rigid-alias-no-params.rs ... ok
---
131    |
- help: you might have meant to use the similarly named previously used binding `c`
+ help: you might have meant to use the similarly named previously used binding `a`
133    |
134 LL -     let (A(A(a, b) | B(c), d) | B(e)) = Y;
- LL +     let (A(A(a, b) | B(c), d) | B(c)) = Y;
+ LL +     let (A(A(a, b) | B(c), d) | B(a)) = Y;
136    |
137 
138 error[E0408]: variable `a` is not bound in all patterns

262 ...
263 LL |               V3(c),
264    |               ^^^^^ pattern doesn't bind `b`
-    |
- help: you might have meant to use the similarly named previously used binding `c`
-    |
- LL -                     B(b),
- LL +                     B(c),
-    |
271 
272 error[E0408]: variable `c` is not bound in all patterns
273   --> $DIR/missing-bindings.rs:59:13


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args or-patterns/missing-bindings.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/or-patterns/missing-bindings.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/or-patterns/missing-bindings" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2018"
stdout: none
--- stderr -------------------------------
error[E0408]: variable `beta` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:19:10
   |
LL |     let (alpha | beta | charlie) = alpha; //~  ERROR variable `beta` is not bound in all patterns
   |          ^^^^^   ----   ^^^^^^^ pattern doesn't bind `beta`
   |          |       |
   |          |       variable not in all patterns
   |          pattern doesn't bind `beta`

error[E0408]: variable `beta` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:22:14
   |
LL |         Some(alpha | beta) => {} //~ ERROR variable `beta` is not bound in all patterns
   |              ^^^^^   ---- variable not in all patterns
   |              |
   |              pattern doesn't bind `beta`

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:35:20
   |
LL |     let (A(a, _) | _) = X; //~ ERROR variable `a` is not bound in all patterns
   |            -       ^ pattern doesn't bind `a`
   |            |
   |            variable not in all patterns

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:36:10
   |
LL |     let (_ | B(a)) = X; //~ ERROR variable `a` is not bound in all patterns
   |          ^     - variable not in all patterns
   |          |
   |          pattern doesn't bind `a`

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:37:10
   |
LL |     let (A(..) | B(a)) = X; //~ ERROR variable `a` is not bound in all patterns
   |          ^^^^^     - variable not in all patterns
   |          |
   |          pattern doesn't bind `a`

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:38:20
   |
LL |     let (A(a, _) | B(_)) = X; //~ ERROR variable `a` is not bound in all patterns
   |            -       ^^^^ pattern doesn't bind `a`
   |            |
   |            variable not in all patterns

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:39:20
   |
LL |     let (A(_, a) | B(_)) = X; //~ ERROR variable `a` is not bound in all patterns
   |               -    ^^^^ pattern doesn't bind `a`
   |               |
   |               variable not in all patterns

error[E0408]: variable `b` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:40:20
   |
LL |     let (A(a, b) | B(a)) = X; //~ ERROR variable `b` is not bound in all patterns
   |               -    ^^^^ pattern doesn't bind `b`
   |               |
   |               variable not in all patterns

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:44:10
   |
LL |     let (A(A(..) | B(_), _) | B(a)) = Y; //~ ERROR variable `a` is not bound in all patterns
   |          ^^^^^^^^^^^^^^^^^^     - variable not in all patterns
   |          |
   |          pattern doesn't bind `a`

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:45:12
   |
LL |     let (A(A(..) | B(a), _) | B(A(a, _) | B(a))) = Y;
   |            ^^^^^     - variable not in all patterns
   |            |
   |            pattern doesn't bind `a`

error[E0408]: variable `c` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:47:12
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |            ^^^^^^^     - variable not in all patterns
   |            |
   |            pattern doesn't bind `c`
   |
help: you might have meant to use the similarly named previously used binding `a`
   |
LL -     let (A(A(a, b) | B(c), d) | B(e)) = Y;
LL +     let (A(A(a, b) | B(a), d) | B(e)) = Y;
   |

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:47:22
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |              -       ^^^^ pattern doesn't bind `a`
   |              |
   |              variable not in all patterns
   |
help: you might have meant to use the similarly named previously used binding `c`
   |
LL -     let (A(A(a, b) | B(c), d) | B(e)) = Y;
LL +     let (A(A(c, b) | B(c), d) | B(e)) = Y;
   |

error[E0408]: variable `b` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:47:22
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |                 -    ^^^^ pattern doesn't bind `b`
   |                 |
   |                 variable not in all patterns
   |
help: you might have meant to use the similarly named previously used binding `c`
   |
LL -     let (A(A(a, b) | B(c), d) | B(e)) = Y;
LL +     let (A(A(a, c) | B(c), d) | B(e)) = Y;
   |

error[E0408]: variable `e` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:47:10
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |          ^^^^^^^^^^^^^^^^^^^^     - variable not in all patterns
   |          |
   |          pattern doesn't bind `e`
   |
help: you might have meant to use the similarly named previously used binding `a`
   |
LL -     let (A(A(a, b) | B(c), d) | B(e)) = Y;
LL +     let (A(A(a, b) | B(c), d) | B(a)) = Y;
   |

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:47:33
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |              -                  ^^^^ pattern doesn't bind `a`
   |              |
   |              variable not in all patterns
   |
help: you might have meant to use the similarly named previously used binding `e`
   |
LL -     let (A(A(a, b) | B(c), d) | B(e)) = Y;
LL +     let (A(A(e, b) | B(c), d) | B(e)) = Y;
   |

error[E0408]: variable `b` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:47:33
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |                 -               ^^^^ pattern doesn't bind `b`
   |                 |
   |                 variable not in all patterns
   |
help: you might have meant to use the similarly named previously used binding `e`
   |
LL -     let (A(A(a, b) | B(c), d) | B(e)) = Y;
LL +     let (A(A(a, e) | B(c), d) | B(e)) = Y;
   |

error[E0408]: variable `c` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:47:33
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |                        -        ^^^^ pattern doesn't bind `c`
   |                        |
   |                        variable not in all patterns
   |
help: you might have meant to use the similarly named previously used binding `e`
   |
LL -     let (A(A(a, b) | B(c), d) | B(e)) = Y;
LL +     let (A(A(a, b) | B(e), d) | B(e)) = Y;
   |

error[E0408]: variable `d` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:47:33
   |
LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
   |                            -    ^^^^ pattern doesn't bind `d`
   |                            |
   |                            variable not in all patterns
   |
help: you might have meant to use the similarly named previously used binding `e`
   |
LL -     let (A(A(a, b) | B(c), d) | B(e)) = Y;
LL +     let (A(A(a, b) | B(c), e) | B(e)) = Y;
   |

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:63:29
   |
LL |                     Ok(a) | Err(_), //~ ERROR variable `a` is not bound in all patterns
   |                        -    ^^^^^^ pattern doesn't bind `a`
   |                        |
   |                        variable not in all patterns

error[E0408]: variable `b` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:70:21
   |
LL |                     A(_, a) | //~ ERROR variable `b` is not bound in all patterns
   |                     ^^^^^^^ pattern doesn't bind `b`
LL |                     B(b), //~ ERROR variable `a` is not bound in all patterns
   |                       - variable not in all patterns
   |
help: you might have meant to use the similarly named previously used binding `a`
   |
LL -                     B(b), //~ ERROR variable `a` is not bound in all patterns
LL +                     B(a), //~ ERROR variable `a` is not bound in all patterns
   |

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:71:21
   |
LL |                     A(_, a) | //~ ERROR variable `b` is not bound in all patterns
   |                          - variable not in all patterns
LL |                     B(b), //~ ERROR variable `a` is not bound in all patterns
   |                     ^^^^ pattern doesn't bind `a`
   |
help: you might have meant to use the similarly named previously used binding `b`
   |
LL -                     A(_, a) | //~ ERROR variable `b` is not bound in all patterns
LL +                     A(_, b) | //~ ERROR variable `b` is not bound in all patterns
   |

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:74:17
   |
LL |                     A(_, a) | //~ ERROR variable `b` is not bound in all patterns
   |                          - variable not in all patterns
...
LL |                 B(_)
   |                 ^^^^ pattern doesn't bind `a`

error[E0408]: variable `b` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:74:17
   |
LL |                     B(b), //~ ERROR variable `a` is not bound in all patterns
   |                       - variable not in all patterns
...
LL |                 B(_)
   |                 ^^^^ pattern doesn't bind `b`

---
LL | |             //~^ ERROR variable `b` is not bound in all patterns
LL | |             //~| ERROR variable `c` is not bound in all patterns
LL | |                 A(
...  |
LL | |                 B(Ok(a) | Err(a))
LL | |             ) |
   | |_____________^ pattern doesn't bind `b`
...
LL |                       B(b), //~ ERROR variable `a` is not bound in all patterns
   |                         - variable not in all patterns
...
LL |               V3(c),
   |               ^^^^^ pattern doesn't bind `b`

---
LL | |             //~^ ERROR variable `b` is not bound in all patterns
LL | |             //~| ERROR variable `c` is not bound in all patterns
LL | |                 A(
...  |
LL | |                 B(Ok(a) | Err(a))
LL | |             ) |
   | |_____________^ pattern doesn't bind `c`
LL | /             V2(
LL | |                 A(
LL | |                     A(_, a) | //~ ERROR variable `b` is not bound in all patterns
LL | |                     B(b), //~ ERROR variable `a` is not bound in all patterns
...  |
LL | |             ) |
   | |_____________^ pattern doesn't bind `c`
LL |               V3(c),
   |                  - variable not in all patterns
---

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:78:13
   |
LL |                 B(Ok(a) | Err(a))
   |                               - variable not in all patterns
...
LL |                     A(_, a) | //~ ERROR variable `b` is not bound in all patterns
   |                          - variable not in all patterns
...
LL |             V3(c),
   |             ^^^^^ pattern doesn't bind `a`
   |
help: you might have meant to use the similarly named previously used binding `c`
   |
LL ~                 B(Ok(a) | Err(c))
LL |             ) |
LL |             V2(
LL |                 A(
LL ~                     A(_, c) | //~ ERROR variable `b` is not bound in all patterns
   |

error[E0170]: pattern binding `beta` is named the same as one of the variants of the type `check_handling_of_paths::bar::foo`
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:19:18
   |
LL |     let (alpha | beta | charlie) = alpha; //~  ERROR variable `beta` is not bound in all patterns
   |                  ^^^^
   |
   = note: `#[deny(bindings_with_variant_name)]` on by default

error[E0170]: pattern binding `beta` is named the same as one of the variants of the type `check_handling_of_paths::bar::foo`
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:22:22
   |
LL |         Some(alpha | beta) => {} //~ ERROR variable `beta` is not bound in all patterns
   |                      ^^^^ help: to match on the variant, qualify the path: `check_handling_of_paths::bar::foo::beta`

error[E0004]: non-exhaustive patterns: `None` not covered
##[error]  --> /checkout/tests/ui/or-patterns/missing-bindings.rs:21:11
   |
LL |     match Some(alpha) { //~ ERROR `None` not covered
   |           ^^^^^^^^^^^ pattern `None` not covered
   |
note: `Option<foo>` defined here
  --> /rustc/FAKE_PREFIX/library/core/src/option.rs:600:0
  ::: /rustc/FAKE_PREFIX/library/core/src/option.rs:604:4
   |
   = note: not covered
   = note: the matched value is of type `Option<foo>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
   |
LL |         Some(alpha | beta) => {}, None => todo!() //~ ERROR variable `beta` is not bound in all patterns
   |                                 +++++++++++++++++

error: aborting due to 29 previous errors

Some errors have detailed explanations: E0004, E0170, E0408.
---
10    |
- help: you might have meant to use the similarly named previously used binding `c`
+ help: you might have meant to use the similarly named previously used binding `a`
12    |
13 LL -         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
- LL +         T::T1(a, d) | T::T2(d, c) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
+ LL +         T::T1(a, d) | T::T2(d, a) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
15    |
16 
17 error[E0408]: variable `c` is not bound in all patterns

24    |         |             pattern doesn't bind `c`
25    |         pattern doesn't bind `c`
26    |
- help: you might have meant to use the similarly named previously used binding `d`
+ help: you might have meant to use the similarly named previously used binding `a`
28    |
29 LL -         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
- LL +         T::T1(a, d) | T::T2(d, b) | T::T3(d) | T::T4(a) => { println!("{:?}", a); }
+ LL +         T::T1(a, d) | T::T2(d, b) | T::T3(a) | T::T4(a) => { println!("{:?}", a); }
31    |
32 
33 error[E0408]: variable `a` is not bound in all patterns

40    |               |       pattern doesn't bind `a`
41    |               variable not in all patterns
42    |
- help: you might have meant to use the similarly named previously used binding `c`
+ help: you might have meant to use the similarly named previously used binding `b`
44    |
45 LL -         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
- LL +         T::T1(c, d) | T::T2(d, b) | T::T3(c) | T::T4(c) => { println!("{:?}", a); }
+ LL +         T::T1(b, d) | T::T2(d, b) | T::T3(c) | T::T4(b) => { println!("{:?}", a); }
47    |
48 
49 error[E0408]: variable `d` is not bound in all patterns

55    |                  |          |       pattern doesn't bind `d`
56    |                  |          variable not in all patterns
57    |                  variable not in all patterns
-    |
- help: you might have meant to use the similarly named previously used binding `c`
-    |
- LL -         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
- LL +         T::T1(a, c) | T::T2(c, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
-    |
64 
65 error[E0381]: used binding `a` is possibly-uninitialized
66   --> $DIR/issue-39698.rs:10:79


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args span/issue-39698.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/span/issue-39698.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/span/issue-39698" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0408]: variable `b` is not bound in all patterns
##[error]  --> /checkout/tests/ui/span/issue-39698.rs:10:9
   |
LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
   |         ^^^^^^^^^^^            -    ^^^^^^^^   ^^^^^^^^ pattern doesn't bind `b`
   |         |                      |    |
   |         |                      |    pattern doesn't bind `b`
   |         |                      variable not in all patterns
   |         pattern doesn't bind `b`
   |
help: you might have meant to use the similarly named previously used binding `a`
   |
LL -         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
LL +         T::T1(a, d) | T::T2(d, a) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
   |

error[E0408]: variable `c` is not bound in all patterns
##[error]  --> /checkout/tests/ui/span/issue-39698.rs:10:9
   |
LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
   |         ^^^^^^^^^^^   ^^^^^^^^^^^         -    ^^^^^^^^ pattern doesn't bind `c`
   |         |             |                   |
   |         |             |                   variable not in all patterns
   |         |             pattern doesn't bind `c`
   |         pattern doesn't bind `c`
   |
help: you might have meant to use the similarly named previously used binding `a`
   |
LL -         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
LL +         T::T1(a, d) | T::T2(d, b) | T::T3(a) | T::T4(a) => { println!("{:?}", a); }
   |

error[E0408]: variable `a` is not bound in all patterns
##[error]  --> /checkout/tests/ui/span/issue-39698.rs:10:23
   |
LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
   |               -       ^^^^^^^^^^^   ^^^^^^^^         - variable not in all patterns
   |               |       |             |
   |               |       |             pattern doesn't bind `a`
   |               |       pattern doesn't bind `a`
   |               variable not in all patterns
   |
help: you might have meant to use the similarly named previously used binding `b`
   |
LL -         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
LL +         T::T1(b, d) | T::T2(d, b) | T::T3(c) | T::T4(b) => { println!("{:?}", a); }
   |

error[E0408]: variable `d` is not bound in all patterns
##[error]  --> /checkout/tests/ui/span/issue-39698.rs:10:37
   |
LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
   |                  -          -       ^^^^^^^^   ^^^^^^^^ pattern doesn't bind `d`
   |                  |          |       |
   |                  |          |       pattern doesn't bind `d`
   |                  |          variable not in all patterns
   |                  variable not in all patterns

error[E0381]: used binding `a` is possibly-uninitialized
##[error]  --> /checkout/tests/ui/span/issue-39698.rs:10:79
   |
LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
   |               -                                      -                        ^ `a` used here but it is possibly-uninitialized
   |               |                                      |
   |               |                                      binding initialized here in some conditions
   |               binding initialized here in some conditions
   |               binding declared here but left uninitialized

---
---- [ui] tests/ui/traits/item-privacy.rs stdout ----
Saved the actual stderr to `/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/traits/item-privacy/item-privacy.stderr`
diff of stderr:

150    |               - this trait is not dyn compatible...
151 LL |         const C: u8 = 0;
152    |               ^ ...because it contains associated const `C`
-    = help: consider moving `C` to another trait
154    = help: consider moving `A` to another trait
155    = help: consider moving `B` to another trait
+    = help: consider moving `C` to another trait
156    = help: only type `S` implements `assoc_const::C`; consider using it directly instead.
157 
158 error[E0038]: the trait `assoc_const::C` is not dyn compatible

175    |               - this trait is not dyn compatible...
176 LL |         const C: u8 = 0;
177    |               ^ ...because it contains associated const `C`
-    = help: consider moving `C` to another trait
179    = help: consider moving `A` to another trait
180    = help: consider moving `B` to another trait
+    = help: consider moving `C` to another trait
181    = help: only type `S` implements `assoc_const::C`; consider using it directly instead.
182 
183 error[E0223]: ambiguous associated type


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args traits/item-privacy.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/traits/item-privacy.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/traits/item-privacy" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2015"
stdout: none
--- stderr -------------------------------
error[E0599]: no method named `a` found for struct `S` in the current scope
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:68:7
   |
LL | struct S;
   | -------- method `a` not found for this struct
...
LL |     S.a(); //~ ERROR no method named `a` found
   |       ^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = help: trait `method::A` which provides `a` is implemented but not reachable
help: there is a method `b` with a similar name
   |
LL -     S.a(); //~ ERROR no method named `a` found
LL +     S.b(); //~ ERROR no method named `a` found
   |

error[E0599]: no method named `b` found for struct `S` in the current scope
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:69:7
   |
LL | struct S;
   | -------- method `b` not found for this struct
...
LL |         fn b(&self) { }
   |            - the method is available for `S` here
...
LL |     S.b(); //~ ERROR no method named `b` found
   |       ^
   |
   = help: items from traits can only be used if the trait is in scope
help: trait `B` which provides `b` is implemented but not in scope; perhaps you want to import it
   |
LL + use method::B;
   |
help: there is a method `c` with a similar name
   |
LL -     S.b(); //~ ERROR no method named `b` found
LL +     S.c(); //~ ERROR no method named `b` found
   |

error[E0624]: method `a` is private
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:73:7
   |
LL |         fn a(&self) { }
   |         ----------- private method defined here
...
LL |     c.a(); //~ ERROR method `a` is private
   |       ^ private method

error[E0599]: no function or associated item named `a` found for struct `S` in the current scope
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:79:8
   |
LL | struct S;
   | -------- function or associated item `a` not found for this struct
...
LL |     S::a(&S);
   |        ^ function or associated item not found in `S`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = help: trait `method::A` which provides `a` is implemented but not reachable
help: there is an associated constant `B` with a similar name
  --> /checkout/tests/ui/traits/item-privacy.rs:30:9
   |
LL |         const B: u8 = 0;
   |         ^^^^^^^^^^^

error[E0599]: no function or associated item named `b` found for struct `S` in the current scope
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:81:8
   |
LL | struct S;
   | -------- function or associated item `b` not found for this struct
...
LL |     S::b(&S);
   |        ^ function or associated item not found in `S`
   |
   = help: items from traits can only be used if the trait is in scope
help: there is an associated constant `B` with a similar name
  --> /checkout/tests/ui/traits/item-privacy.rs:30:9
   |
LL |         const B: u8 = 0;
   |         ^^^^^^^^^^^
help: trait `B` which provides `b` is implemented but not in scope; perhaps you want to import it
   |
LL + use method::B;
   |

error[E0624]: method `a` is private
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:85:14
   |
LL |         fn a(&self) { }
   |         ----------- private method defined here
...
LL |     <dyn C>::a(&S); //~ ERROR method `a` is private
   |              ^ private method

error[E0599]: no associated item named `A` found for struct `S` in the current scope
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:98:8
   |
LL | struct S;
   | -------- associated item `A` not found for this struct
...
LL |     S::A; //~ ERROR no associated item named `A` found
   |        ^ associated item not found in `S`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = help: trait `assoc_const::A` which provides `A` is implemented but not reachable
help: there is an associated constant `B` with a similar name
   |
LL -     S::A; //~ ERROR no associated item named `A` found
LL +     S::B; //~ ERROR no associated item named `A` found
   |

error[E0599]: no associated item named `B` found for struct `S` in the current scope
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:99:8
   |
LL | struct S;
   | -------- associated item `B` not found for this struct
...
LL |     S::B; //~ ERROR no associated item named `B` found
   |        ^ associated item not found in `S`
   |
   = help: items from traits can only be used if the trait is in scope
help: there is a method `b` with a similar name
  --> /checkout/tests/ui/traits/item-privacy.rs:12:9
   |
LL |         fn b(&self) { }
   |         ^^^^^^^^^^^
help: trait `B` which provides `B` is implemented but not in scope; perhaps you want to import it
   |
LL + use assoc_const::B;
   |

error[E0038]: the trait `assoc_const::C` is not dyn compatible
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:102:10
   |
LL |     <dyn C>::A;
   |          ^ `assoc_const::C` is not dyn compatible
   |
note: for a trait to be dyn compatible it needs to allow building a vtable
      for more information, visit <https://doc.rust-lang.org/reference/items/traits.html#dyn-compatibility>
  --> /checkout/tests/ui/traits/item-privacy.rs:26:15
   |
LL |         const A: u8 = 0;
   |               ^ ...because it contains associated const `A`
...
LL |         const B: u8 = 0;
   |               ^ ...because it contains associated const `B`
...
LL |     pub trait C: A + B {
   |               - this trait is not dyn compatible...
LL |         const C: u8 = 0;
   |               ^ ...because it contains associated const `C`
   = help: consider moving `A` to another trait
   = help: consider moving `B` to another trait
   = help: consider moving `C` to another trait
   = help: only type `S` implements `assoc_const::C`; consider using it directly instead.

error[E0038]: the trait `assoc_const::C` is not dyn compatible
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:104:10
   |
LL |     <dyn C>::B;
   |          ^ `assoc_const::C` is not dyn compatible
   |
note: for a trait to be dyn compatible it needs to allow building a vtable
      for more information, visit <https://doc.rust-lang.org/reference/items/traits.html#dyn-compatibility>
  --> /checkout/tests/ui/traits/item-privacy.rs:26:15
   |
LL |         const A: u8 = 0;
   |               ^ ...because it contains associated const `A`
...
LL |         const B: u8 = 0;
   |               ^ ...because it contains associated const `B`
...
LL |     pub trait C: A + B {
   |               - this trait is not dyn compatible...
LL |         const C: u8 = 0;
   |               ^ ...because it contains associated const `C`
   = help: consider moving `A` to another trait
   = help: consider moving `B` to another trait
   = help: consider moving `C` to another trait
   = help: only type `S` implements `assoc_const::C`; consider using it directly instead.

error[E0223]: ambiguous associated type
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:117:12
   |
LL |     let _: S::A; //~ ERROR ambiguous associated type
   |            ^^^^
   |
help: if there were a trait named `Example` with associated type `A` implemented for `S`, you could use the fully-qualified path
   |
LL -     let _: S::A; //~ ERROR ambiguous associated type
LL +     let _: <S as Example>::A; //~ ERROR ambiguous associated type
   |

error[E0223]: ambiguous associated type
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:118:12
   |
LL |     let _: S::B; //~ ERROR ambiguous associated type
   |            ^^^^
   |
help: use fully-qualified syntax
   |
LL -     let _: S::B; //~ ERROR ambiguous associated type
LL +     let _: <S as assoc_ty::B>::B; //~ ERROR ambiguous associated type
   |

error[E0223]: ambiguous associated type
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:119:12
   |
LL |     let _: S::C; //~ ERROR ambiguous associated type
   |            ^^^^
   |
help: use fully-qualified syntax
   |
LL -     let _: S::C; //~ ERROR ambiguous associated type
LL +     let _: <S as assoc_ty::C>::C; //~ ERROR ambiguous associated type
   |

error[E0624]: associated type `A` is private
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:121:12
   |
LL |         type A = u8;
   |         ------ the associated type is defined here
...
LL |     let _: T::A; //~ ERROR associated type `A` is private
   |            ^^^^ private associated type

error[E0624]: associated type `A` is private
##[error]  --> /checkout/tests/ui/traits/item-privacy.rs:130:9
   |
LL |         type A = u8;
   |         ------ the associated type is defined here
...
LL |         A = u8, //~ ERROR associated type `A` is private
   |         ^^^^^^ private associated type

error: aborting due to 15 previous errors

Some errors have detailed explanations: E0038, E0223, E0599, E0624.
```

</details>

--- a/tests/ui/or-patterns/missing-bindings.stderr
+++ b/tests/ui/or-patterns/missing-bindings.stderr
@@ -129,10 +129,10 @@ LL |     let (A(A(a, b) | B(c), d) | B(e)) = Y;
    |          |
    |          pattern doesn't bind `e`
    |
-help: you might have meant to use the similarly named previously used binding `c`
+help: you might have meant to use the similarly named previously used binding `a`
    |
 LL -     let (A(A(a, b) | B(c), d) | B(e)) = Y;
-LL +     let (A(A(a, b) | B(c), d) | B(c)) = Y;
+LL +     let (A(A(a, b) | B(c), d) | B(a)) = Y;
    |
 
 error[E0408]: variable `a` is not bound in all patterns
@@ -262,12 +262,6 @@ LL |                       B(b),
 ...
 LL |               V3(c),
    |               ^^^^^ pattern doesn't bind `b`
-   |
-help: you might have meant to use the similarly named previously used binding `c`
-   |
-LL -                     B(b),
-LL +                     B(c),
-   |
 
 error[E0408]: variable `c` is not bound in all patterns
   --> $DIR/missing-bindings.rs:59:13

--- a/tests/ui/span/issue-39698.stderr
+++ b/tests/ui/span/issue-39698.stderr
@@ -8,10 +8,10 @@ LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}
    |         |                      variable not in all patterns
    |         pattern doesn't bind `b`
    |
-help: you might have meant to use the similarly named previously used binding `c`
+help: you might have meant to use the similarly named previously used binding `a`
    |
 LL -         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
-LL +         T::T1(a, d) | T::T2(d, c) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
+LL +         T::T1(a, d) | T::T2(d, a) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
    |
 
 error[E0408]: variable `c` is not bound in all patterns
@@ -24,10 +24,10 @@ LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}
    |         |             pattern doesn't bind `c`
    |         pattern doesn't bind `c`
    |
-help: you might have meant to use the similarly named previously used binding `d`
+help: you might have meant to use the similarly named previously used binding `a`
    |
 LL -         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
-LL +         T::T1(a, d) | T::T2(d, b) | T::T3(d) | T::T4(a) => { println!("{:?}", a); }
+LL +         T::T1(a, d) | T::T2(d, b) | T::T3(a) | T::T4(a) => { println!("{:?}", a); }
    |
 
 error[E0408]: variable `a` is not bound in all patterns
@@ -40,10 +40,10 @@ LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}
    |               |       pattern doesn't bind `a`
    |               variable not in all patterns
    |
-help: you might have meant to use the similarly named previously used binding `c`
+help: you might have meant to use the similarly named previously used binding `b`
    |
 LL -         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
-LL +         T::T1(c, d) | T::T2(d, b) | T::T3(c) | T::T4(c) => { println!("{:?}", a); }
+LL +         T::T1(b, d) | T::T2(d, b) | T::T3(c) | T::T4(b) => { println!("{:?}", a); }
    |
 
 error[E0408]: variable `d` is not bound in all patterns
@@ -55,12 +55,6 @@ LL |         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}
    |                  |          |       pattern doesn't bind `d`
    |                  |          variable not in all patterns
    |                  variable not in all patterns
-   |
-help: you might have meant to use the similarly named previously used binding `c`
-   |
-LL -         T::T1(a, d) | T::T2(d, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
-LL +         T::T1(a, c) | T::T2(c, b) | T::T3(c) | T::T4(a) => { println!("{:?}", a); }
-   |
 
 error[E0381]: used binding `a` is possibly-uninitialized
   --> $DIR/issue-39698.rs:10:79

--- a/tests/ui/traits/item-privacy.stderr
+++ b/tests/ui/traits/item-privacy.stderr
@@ -150,9 +150,9 @@ LL |     pub trait C: A + B {
    |               - this trait is not dyn compatible...
 LL |         const C: u8 = 0;
    |               ^ ...because it contains associated const `C`
-   = help: consider moving `C` to another trait
    = help: consider moving `A` to another trait
    = help: consider moving `B` to another trait
+   = help: consider moving `C` to another trait
    = help: only type `S` implements `assoc_const::C`; consider using it directly instead.
 
 error[E0038]: the trait `assoc_const::C` is not dyn compatible
@@ -175,9 +175,9 @@ LL |     pub trait C: A + B {
    |               - this trait is not dyn compatible...
 LL |         const C: u8 = 0;
    |               ^ ...because it contains associated const `C`
-   = help: consider moving `C` to another trait
    = help: consider moving `A` to another trait
    = help: consider moving `B` to another trait
+   = help: consider moving `C` to another trait
    = help: only type `S` implements `assoc_const::C`; consider using it directly instead.
 
 error[E0223]: ambiguous associated type

